src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -512,19 +515,25 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             {
                 sendTask = base.SendAsync(request, cts.Token);
             }
+            catch (OperationCanceledException e) when (TimeoutFired(cancellationToken, timeoutTime))
+            {
+                HandleFinishSendAsyncCleanup(cts, disposeCts);
+                ThrowTimeoutException(e);
+                throw;
+            }
             catch
             {
                 HandleFinishSendAsyncCleanup(cts, disposeCts);
```

Comment:
Alternatively, you can avoid the multiple catch blocks and the catch block filter:
```C#
catch (Exception e)
{
    HandleFinishSendAsyncCleanup(cts, disposeCts);
    if (e is OperationCanceledException && TimeoutFired(cancellationToken, timeoutTime))
    {
        ThrowTimeoutException(e);
    }
    throw;
}
```

Reply:
I thought it'd be better to clearly separate logic of timeout handling since the amount of duplication is quite small. However, I have changed the code as you proposed since you say that it's better to avoid duplication and exception filters.